fix(ast): add missing estree props for `ArrowFunctionExpression`

camchenry opened the pull request:
building on @overlookmotel and @hi-ogawa's fine work, I thought I would lend a hand at trying to add some missing estree props.

this PR adds the `generator` property and `id` property for `ArrowFunctionExpression`. the `id` property requires serializing a constant `null` value, which seemed like it required some funky output in the serializer. would appreciate some suggestions on how to make this cleaner 🙏

camchenry commented:
* **#8980** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8980?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/8980?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=8980).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hi-ogawa commented on crates/oxc_ast/src/ast/js.rs:
```diff
@@ -1709,6 +1709,7 @@ pub struct FunctionBody<'a> {
 )]
 #[derive(Debug)]
 #[generate_derive(CloneIn, GetSpan, GetSpanMut, ContentEq, ESTree)]
+#[estree(add_fields(generator = false, id = null))]
```
It might be okay to write it directly as rust expression? Also `add_ts` is required for .d.ts output.

```suggestion
#[estree(add_fields(generator = false, id = Option::<()>::None), add_ts="generator: false; id: null;")]
```

overlookmotel replied:
Thanks for getting involved @camchenry.

`()` serializes to `null`, so that's probably the simplest way:

```rs
#[estree(add_fields(generator = false, id = (), add_ts="generator: false; id: null;")]
```

camchenry replied:
Oh perfect, I think adding an alias here makes a lot of sense. It's intuitive to me that `None` would serialize to that, but it's not easy to use everywhere, but `()` is better.

overlookmotel commented:
### Merge activity

* **Feb 10, 8:12 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
fix(ast): add missing estree props for `ArrowFunctionExpression` (#8980)

building on @overlookmotel and @hi-ogawa's fine work, I thought I would lend a hand at trying to add some missing estree props.

this PR adds the `generator` property and `id` property for `ArrowFunctionExpression`. the `id` property requires serializing a constant `null` value, which seemed like it required some funky output in the serializer. would appreciate some suggestions on how to make this cleaner 🙏
